qiita_db/processing_job.py

```diff
@@ -620,6 +620,29 @@ def _update_and_launch_children(self, mapping):
         for c in ready:
             c.submit()
 
+    @property
+    def outputs(self):
+        """The outputs of the job
+
+        Returns
+        -------
+        dict of {str: qiita_db.artifact.Artifact}
+            The outputs of the job keyed by output name
+        """
+        with qdb.sql_connection.TRN:
+            if self.status != 'success':
```

Comment:
It is tested here: https://github.com/biocore/qiita/pull/1933/files#diff-0e7761c945e8eddac7dcc4afcb8f1a35R437

The only reason why I preferred to raise an error is to differentiate the case in which the job did not finish from the cases in which (1) the job did not generate any output (not sure if this will ever happen) or (2) the output artifact has been removed